integration/docker/vsock_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package docker
+
+import (
+	"os"
+
+	"github.com/kata-containers/tests"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("vsock test", func() {
+	var (
+		args     []string
+		id       string = randomDockerName()
+		exitCode int
+		stdout   string
+	)
+
+	AfterEach(func() {
+		Expect(RemoveDockerContainer(id)).To(BeTrue())
+		Expect(ExistDockerContainer(id)).NotTo(BeTrue())
+	})
+
+	Context("when using vsock", func() {
+		It("should not create a kata-proxy process", func() {
+			val, ok := os.LookupEnv("USE_VSOCK")
+			if !ok {
+				Skip("Use of vsock not enabled")
+			}
+			if val != "yes" {
+				Skip("Use of vsock not enabled")
+			}
+			args = []string{"--name", id, "-d", Image, "top"}
+			_, _, exitCode = dockerRun(args...)
+			Expect(exitCode).To(Equal(0))
+
+			// Check no kata-proxy process is running
```

Comment:
Do not use `ps`, use https://github.com/kata-containers/tests/blob/master/process.go#L66 instead